src/binder_cbs.c

```diff
@@ -180,50 +191,85 @@ binder_cbs_set_config(
-        const char* entry = list[i];
-        const char* delim = strchr(entry, '-');
-
-        config->selected = TRUE;
-        config->toCodeScheme = 0xff;
-        if (delim) {
-            char** range = g_strsplit(entry, "-", 0);
-
-            config->fromServiceId = atoi(range[0]);
-            config->toServiceId = atoi(range[1]);
-            g_strfreev(range);
-        } else {
-            config->fromServiceId = config->toServiceId = atoi(entry);
+
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        /* setGsmBroadcastConfig(int32_t serial, vec<GsmBroadcastSmsConfigInfo>); */
+        GBinderParent parent;
+        GBinderHidlVec* vec = gbinder_writer_new0(&writer, GBinderHidlVec);
+        RadioGsmBroadcastSmsConfig* configs = NULL;
+        guint i;
+
+        vec->count = count;
+        vec->owns_buffer = TRUE;
+        vec->data.ptr = configs = gbinder_writer_malloc0(&writer,
+            sizeof(RadioGsmBroadcastSmsConfig) * count);
+
+        for (i = 0; i < count; i++) {
+            RadioGsmBroadcastSmsConfig* config = configs + i;
+            const char* entry = list[i];
+            const char* delim = strchr(entry, '-');
+
+            config->selected = TRUE;
+            config->toCodeScheme = 0xff;
+            if (delim) {
+                char** range = g_strsplit(entry, "-", 0);
+
+                config->fromServiceId = atoi(range[0]);
+                config->toServiceId = atoi(range[1]);
+                g_strfreev(range);
+            } else {
+                config->fromServiceId = config->toServiceId = atoi(entry);
+            }
         }
-    }
 
-    /* Every vector, even the one without data, requires two buffer objects */
-    parent.offset = GBINDER_HIDL_VEC_BUFFER_OFFSET;
-    parent.index = gbinder_writer_append_buffer_object(&writer, vec,
-        sizeof(*vec));
-    gbinder_writer_append_buffer_object_with_parent(&writer, configs,
-        sizeof(configs[0]) * count, &parent);
+        /* Every vector, even the one without data, requires two buffer objects */
+        parent.offset = GBINDER_HIDL_VEC_BUFFER_OFFSET;
+        parent.index = gbinder_writer_append_buffer_object(&writer, vec,
+            sizeof(*vec));
+        gbinder_writer_append_buffer_object_with_parent(&writer, configs,
+            sizeof(configs[0]) * count, &parent);
+    } else {
+        /* setGsmBroadcastConfig(int32_t serial, GsmBroadcastSmsConfigInfo[]); */
+        guint i;
+
+        gbinder_writer_append_int32(&writer, count);
+
+        for (i = 0; i < count; i++) {
+            const char* entry = list[i];
+            const char* delim = strchr(entry, '-');
+
+            /* Non-null parcelable */
+            gbinder_writer_append_int32(&writer, 1);
+            /* Parcelable size */
+            gbinder_writer_append_int32(&writer, 6 * sizeof(gint32));
+
+            if (delim) {
+                char** range = g_strsplit(entry, "-", 0);
```

Comment:
Is it guaranteed that each entry will have another entry after `-`? Or does it require checks here.

Reply:
The data is coming from ofono and code there sets either a single value if min and max are same or a range delimited with - if min and max are different so there is always one value on each side of -.

Reply:
Right. So this is a fixed one. OK.